doc/dev/kdb-operations.md

```diff
@@ -0,0 +1,159 @@
+
+Other restrictions:
+
+- Creating mountpoints for root keys (`system:/`, `user:/`, etc.) is not allowed.
+  These parts of the KDB always use hardcoded backends that can only be configured at compile-time.
+- Creating a mountpoint for `system:/elektra` or any mountpoint below is not allowed.
+  This part of the KDB will always be stored in the bootstrap backend.
+  While the only part that is actually required for the bootstrap process is the one below `system:/elektra/mountpoints`, everything below `system:/elektra` is reserved and should only be used internally.
+  Additionally, `system:/elektra/version` and `system:/elektra/modules` will always point to hardcoded read-only backends containing information about this Elektra installation.
+
+## `get` Operation
+
+The purpose of the `get` operation is to read data stored in backends into a `KDB` instance.
+
+> **Note:** Some details of a `get` operation are defined in the [contract with backend plugins](./backend-plugins.md).
+
+Properties of `kdbGet()`:
+
+- After calling `kdbGet (kdb, ks, parentKey)`, the KeySet `ks` will contain _all keys_ (including their values) that are stored in _any backend_ with a mountpoint that is _below `parentKey`_.
+- After calling `kdbGet (kdb, ks, parentKey)`, below `parentKey` the KeySet `ks` will _only_ contain keys that are stored in a backend.
+- The KeySet `ks` may contain other keys not below `parentKey`.
+  These keys fall into one of two categories:
+  1. Keys that are stored in a backend that is not below `parentKey`.
+     `kdbGet()` may decide that it is more efficient (e.g. because of a cache) to return more keys than requested.
+  2. Keys that were already present in `ks` when `kdbGet()` was called.
+
+In simpler terms, this means to the caller it looks as if `kdbGet()` had removed all keys below `parentKey` from `ks` and then loaded the data from the backends.
+Below `parentKey` the KeySet `ks` correctly represents the state of the KDB.
+For the rest of `ks` there are no such guarantees.
+
+TODO: cache correct? notifications? (just call at the end?)
+
+The basic flow of this operation is:
+
+1. Determine the backends needed to read all keys below `parentKey`
+2. Run the `init` phase on all the backends that haven't been initialized
+3. Run the `resolver` phase on all backends
+4. From now on ignore all backends, which indicated that there is no update.
+5. If all backends are now ignored, **return**.
+6. If a global cache plugin is enabled:
+   Ask the global cache plugin for the cache times for all backends.
+7. If all backends have an existing cache entry:
+   Run the `cachecheck` phase on all backends
+8. If all backends indicated the cache is still valid:
+   Ask the global cache plugin for the cached data and **return**.
+9. Run the `prestorage` and `storage` phase on all backends.
+10. Run the `poststorage` phase of all `spec:/` backends.
+11. Merge the data from all backends
+12. If enabled, run the `gopts` plugin.
```

Comment:
Because `gopts` needs access to the `spec:/` namespace. A `proc:/` backend would only have access to `proc:/`.

We could of course call all `spec:/` backends first and then give the corresponding backends in other namespaces access to `spec:/`, but that doesn't match the current isolation.